app/src/androidTest/java/com/willowtree/vocable/PhrasesUseCaseTest.kt

```diff
@@ -152,9 +157,47 @@ class PhrasesUseCaseTest {
                     localizedUtterance = testLocalesWithText,
                     sortOrder = 0,
                     lastSpokenDate = null,
+                    parentCategoryId = "category"
                 )
             ),
             useCase.getPhrasesForCategory("category")
         )
     }
+
+    @Test
+    fun updatePhrase_updatesCustomPhrase() = runTest {
+        val useCase = createUseCase()
+        val phraseId = "1"
+        val localizedUtterance = LocalesWithText(mapOf("en" to "updated text"))
+        storedPhrasesRepository.addPhrase(
+            PhraseDto(
+                phraseId = phraseId,
+                parentCategoryId = "category",
+                creationDate = 0L,
+                lastSpokenDate = null,
+                localizedUtterance = testLocalesWithText,
+                sortOrder = 0
+            )
+        )
+
+        useCase.updatePhrase(phraseId, localizedUtterance)
+
+        val updatedPhrase = useCase.getPhrasesForCategory("category")
+            .single() as CustomPhrase
+        assertEquals(localizedUtterance, updatedPhrase.localizedUtterance)
+    }
+
+    @Test
+    fun updatePhrase_updatesPresetPhrase() = runTest {
+        val useCase = createUseCase()
+        presetPhrasesRepository.populateDatabase()
+        val phraseId = "category_123_0"
+        val localizedUtterance = LocalesWithText(mapOf("en" to "updated text"))
+
+        useCase.updatePhrase(phraseId, localizedUtterance)
+
+        val newCustomPhrase = useCase.getPhrasesForCategory(PresetCategories.USER_KEYPAD.id)
+            .singleOrNull { it.phraseId == phraseId } as CustomPhrase
```

Comment:
Nice! Is this `single` how you're asserting that the preset phrase is no longer being returned?

Reply:
Correct- It was more like the `single` was me assuming/asserting there was only one phrase being returned, and that the one being returned has a localizedUtterance (was not a preset)

Changed it to be a little bit more explicit- using an assert to make it easier to track down test failures on this

Reply:
Nice, I like the explicit assertion, I think that does make it more clear!